graph/graph.go

```diff
@@ -63,11 +69,19 @@ func New(ctx context.Context, choice Subsets) (*DB, error) {
 		}
 	}
 
+	var o driver.Observation
+	if choice.Observation {
+		if o, ok = cfg.Driver.(driver.Observation); !ok {
+			return nil, errors.New("configured driver does not implement observation subset")
+		}
+	}
+
 	return &DB{
 		cfg.Driver,
 		c,
 		h,
 		i,
+		o,
```

Comment:
would consider switching these to codelistDriver, hierarchyDriver etc. If this script balloons it's not gonna be obvious what these are.